### ISSUE TITLE
Pytest 7.0.0 appears to break pytest-pythonpath python_paths in pytest.ini

### ISSUE THREAD

**Issue by bluetech** [pytest-dev/pytest#9636]
### Discussed in https://github.com/pytest-dev/pytest/discussions/9635

<div type='discussions-op-text'>

<sup>Originally posted by **enewton** February  7, 2022</sup>
We have some utility helper code in a separate folder to our tests. Up until now we have this configured via pytest.ini like so:

```
[pytest]
python_paths = ./utils/
```

This definition appears to be getting ignored after upgrading to pytest 7.0.0, which then gives us 'module not found' errors on tests trying to import utils. Reading around, it looks like this functionality is now built in to pytest and so is pytest-pythonpath obsolete?

I can get our tests passing again by updating to:

```
[pytest]
pythonpath = ./utils/
```

So the main question is, is this change in behaviour expected? Has the old method been deliberately disabled or should I raise a bug?
</div>

**Comment by bluetech** [comment]
The problem is that pytest-pythonpath plugin name is `pythonpath`, and the new pytest builtin plugin is also called `pythonpath`. Since the builtin `pythonpath` plugin is registered first, the `pytest-pythonpath` plugin gets ignored.

This was not intentional. We will need to change the builtin plugin's name, probably `python_path`. That's not as nice, but what can you do.

**Comment by The-Compiler** [comment]
Another possible solution would perhaps be to add support for `python_paths` in the core, but deprecate it immediately. That way, it's not a silent failure, but people know that they could get rid of the plugin.

**Comment by bluetech** [comment]
@The-Compiler I like this idea. IIRC there are some minor behavior differences, I will need to review them but if they don't matter then I think we should do it (or make `python_paths` behave compatibly).

**Comment by The-Compiler** [comment]
I don't like the idea of having two names for the same thing in the long run - that sounds like a recipe for confusion.

**Comment by JakNowy** [comment]
I can confirm we're having similar issue in our project. Running pytest test command returns an error:
`ImportError: No module named 'tests.settings'`

We also have tests in a separate directory and pinning pytest version to <7.0 fixes the issue temporary.

**Comment by nicoddemus** [comment]
>  We will need to change the builtin plugin's name, probably to python_path. That's not as nice, but what can you do.

Just to give my 2 cents, I would rather we go with that route: this will fix the problem immediately, without creating future churn for us, and without introducing confusion (like duplicated names). Of course this type of problem (introducing a new core plugin with the same name of an external plugin by accident) might happen again in the future, but regardless this type of problem needs to be fixed/considered on a case-by-case basis.

(Sorry for just jumping in now, after #9640 was already proposed)

**Comment by bluetech** [comment]
I think the deprecations are not too bad because they also serve to tell people "hey you don't need pytest-pythonpath any more".

I believe the logging plugin was once separate. Do you remember how its promotion was handled and if it went well?

**Comment by nicoddemus** [comment]
> I think the deprecations are not too bad because they also serve to tell people "hey you don't need pytest-pythonpath any more".

True, but not sure it is worth the hassle on our side, and who knows if the author of that plugin wants to add further capabilities in the future (but I concede that's unlikely)? The deprecation approach basically "kills" the plugin.  

> Do you remember how its promotion was handled and if it went well?

IIRC we issued a warning if it was installed, but we also released a new pytest-logging with an "empty" package so there wasn't conflicts. In that case it was a concerted effort to integrate them, which is not the same situation with `pytest-pythonpath`.
